From 3073bb87d24c83d9b32dc1a72706885e4f317a05 Mon Sep 17 00:00:00 2001
From: Daniel Tjandra <danieltjandra@gmail.com>
Date: Sun, 8 Jun 2014 23:15:18 -0400
Subject: [PATCH] I9000M froyo circle battery colors

---
 .../android/systemui/BatteryCircleMeterView.java   | 42 ++++++++++++++++++----
 1 file changed, 36 insertions(+), 6 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/BatteryCircleMeterView.java b/packages/SystemUI/src/com/android/systemui/BatteryCircleMeterView.java
index 889ce3f..b8bfb8d 100644
--- a/packages/SystemUI/src/com/android/systemui/BatteryCircleMeterView.java
+++ b/packages/SystemUI/src/com/android/systemui/BatteryCircleMeterView.java
@@ -87,6 +87,8 @@
     private Paint   mPaintGray;
     private Paint   mPaintSystem;
     private Paint   mPaintRed;
+	private Paint	mPaintYellow;
+	private Paint	mPaintGreen;
     private String  mCircleBatteryView;
 
     private int mCircleColor;
@@ -196,9 +198,18 @@ private void drawCircle(Canvas canvas, int level, int animOffset, float textX, R
         Paint usePaint = mPaintSystem;
 
         // turn red when battery is at warning level - same level android battery warning appears
-        if (level <= mWarningLevel) {
+        if (level <= 15) 
+		{
             usePaint = mPaintRed;
         }
+		else if (level > 15 && level <= 50)
+		{
+			usePaint = mPaintYellow;
+		}
+		else if (level > 50)
+		{
+			usePaint = mPaintGreen;
+		}
         usePaint.setAntiAlias(true);
         usePaint.setPathEffect(null);
 
@@ -216,17 +227,30 @@ private void drawCircle(Canvas canvas, int level, int animOffset, float textX, R
         canvas.drawArc(drawRect, 270 + animOffset, 3.6f * padLevel, false, usePaint);
         // if chosen by options, draw percentage text in the middle
         // always skip percentage when 100, so layout doesnt break
-        if (level < 100 && mCirclePercent) {
-            if (level <= mWarningLevel) {
+        if (level < 100 && mCirclePercent) 
+		{
+            if (level <= 15) 
+			{
                 mPaintFont.setColor(mPaintRed.getColor());
-            } else if (mIsCharging) {
+            }
+			else if (level > 15 && level <= 50)
+			{
+				 mPaintFont.setColor(mPaintYellow.getColor());
+			}
+			else if (level > 50)
+			{
+				 mPaintFont.setColor(mPaintGreen.getColor());
+			}
+			else if (mIsCharging) 
+			{
                 mPaintFont.setColor(mCircleTextChargingColor);
-            } else {
+            } 
+			else 
+			{
                 mPaintFont.setColor(mCircleTextColor);
             }
             canvas.drawText(Integer.toString(level), textX, mTextY, mPaintFont);
         }
-
     }
 
     @Override
@@ -291,11 +315,15 @@ private void initializeCircleVars() {
         mPaintGray = new Paint(mPaintFont);
         mPaintSystem = new Paint(mPaintFont);
         mPaintRed = new Paint(mPaintFont);
+		mPaintYellow = new Paint(mPaintFont);
+		mPaintGreen = new Paint(mPaintFont);
 
         // could not find the darker definition anywhere in resources
         // do not want to use static 0x404040 color value. would break theming.
         mPaintGray.setColor(res.getColor(R.color.darker_gray));
         mPaintRed.setColor(res.getColor(R.color.holo_red_light));
+		mPaintYellow.setColor(Color.rgb(255, 255, 0));
+		mPaintGreen.setColor(Color.rgb(0, 255, 0));
 
         // font needs some extra settings
         mPaintFont.setTextAlign(Align.CENTER);
@@ -346,6 +374,8 @@ private void initSizeBasedStuff() {
 
         float strokeWidth = mCircleSize / 7f;
         mPaintRed.setStrokeWidth(strokeWidth);
+        mPaintYellow.setStrokeWidth(strokeWidth);
+        mPaintGreen.setStrokeWidth(strokeWidth);
         mPaintSystem.setStrokeWidth(strokeWidth);
         mPaintGray.setStrokeWidth(strokeWidth / 3.5f);
         // calculate rectangle for drawArc calls
